src/setupTests.js

```diff
@@ -0,0 +1,7 @@
+global.console.warn = message => {
+  throw message;
+};
+
+global.console.error = message => {
+  throw message;
+};
```

Comment:
Pre-commit already used to fail on error, why is this needed?

Reply:
I just included it because of the documentation I was looking at but you're right it is redundant. It has been removed on the most recent push

Reply:
I just included it because of the documentation I was looking at but you're right it is redundant. It has been removed on the most recent push

Reply:
Thanks. without this definition, how does the pre-commit knew until now to block the commits with errors?

Reply:
Its because jest automatically fails tests if they throw errors and the pre-commit blocks the commit if any tests fail

Reply:
Post discussion on this matter:
Currently, pre-commit fails because jest is throwing an exception with exit code 1 in case a test fail. 
This PR is enhancing this behavior so the pre-commit will fail not just for exceptions but for `console.error` and `console.warnings` messages.